caching.md

```diff
@@ -0,0 +1,33 @@
+# Caching Semantics
+
+## Invoked from a GitHub workflow
+
+We cache the contents of `~/.cache/trunk/` if `inputs.cache === true`.
+
+Note that the GitHub Actions cache only caches results on a per-branch/per-PR basis, so the first
+time `Trunk Check` runs on a given branch, it will be uncached. Subsequent runs will read from
+cache.
+
+## Invoked inside the `.trunk` repo
+
+We only cache the contents of `~/.cache/trunk/tools/ruby/`.
```

Comment:
I assume this is because ruby is the biggest/takes the longest to install, or somesuch?